src/imagery/i.landsat/i.landsat.download/i.landsat.download.py

```diff
@@ -206,108 +492,74 @@ def main():
-                )
+        # Should use other provider other than USGS,
+        # as there was a bug in USGS API, fixed here
+        # https://github.com/CS-SI/eodag/issues/1252
+        # Currently availble in EODAG 3.0.0b3 (not yet released)
+        downloader.set_scenes_ids(options["id"].split(","))
+        # TODO: Remove when deserialize_and_register bug is fix
+        if options["datasource"] != "usgs":
+            downloader.download(outdir)
+        return 0
+
+    # Translate sort option to the i.eodag sort options
+    eodag_sort = ""
+    for sort_var in options["sort"].split(","):
+        if sort_var == "cloud_cover":
+            eodag_sort += "cloudcover,"
+        if sort_var == "acquisition_date":
+            eodag_sort += "ingestiondate,"
+
+    # Searching is necessary before listing or downloading
+    downloader.search(
+        dataset=options["dataset"],
+        start=start_date,
+        end=end_date,
+        clouds=options["clouds"] if options["clouds"] else None,
+        vector_map=options["map"] if options["map"] else None,
+        limit=options["limit"],
+        eodag_sort=eodag_sort,
+        eodag_order=options["order"],
+    )
+
+    if flags["l"]:
+        downloader.list()
+        gs.message(
+            _(
+                "To download all scenes found, re-run the previous "
+                "command without -l flag. Note that if no output "
+                "option is provided, files will be downloaded in /tmp"
             )
-
-        else:
-
-            ee = EarthExplorer(user, password)
-
-            for scene in sorted_scenes:
-
-                gs.message(_("Downloading scene <{}> ...").format(scene["entity_id"]))
-
-                ee.download(
-                    identifier=scene["entity_id"],
-                    output_dir=outdir,
-                    timeout=int(options["timeout"]),
-                )
-
-            ee.logout()
+        )
+    else:
+        downloader.download(outdir)
 
 
 if __name__ == "__main__":
     options, flags = gs.parser()
-
-    # lazy import
     try:
-        import landsatxplore.api
-        from landsatxplore.earthexplorer import EarthExplorer
-
-    except ImportError:
-
-        gs.fatal(_("Cannot import landsatxplore. Please install the library first."))
-
-    main()
+        import eodag
+        from eodag import EODataAccessGateway
+    except:
+        gs.fatal(_("Cannot import eodag. Please intall the library first."))
+
+    # Check if eodag config file is created
+    file_path = os.path.join(os.path.expanduser("~"), ".config/eodag/eodag.yml")
+    if not os.path.isfile(file_path):
+        dag = EODataAccessGateway()
+        gs.info(
+            _(
+                "EODAG Config file is created, you can rerun the module after filling the necessary credentials in {}".format(
+                    file_path
+                )
+            )
+        )
+        sys.exit(0)
```

Comment:
Generally the config file is necessary especially when using USGS.
For Planetary Computer, you can go fine most of the time without credentials.

We can remove that exit part, but it will give an error that the user needs to fill in the USGS credentials (if they were attempting to use USGS).


> This would be a show stopper if it is used in e.g. a cloud setting where fressh pod are started to perform downloading...

I see, you are right that would be annoying if it could just go fine without the credentials, and we force the user to rerun again in a different pod. I will remove it for now... Also, I will put the `settings` option back to i.landsat.download once I support it in i.eodag...